app/src/common/shared/org/mozilla/vrbrowser/browser/engine/Session.java

```diff
@@ -0,0 +1,1130 @@
+/* -*- Mode: Java; c-basic-offset: 4; tab-width: 4; indent-tabs-mode: nil; -*-
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.vrbrowser.browser.engine;
+
+import android.app.Activity;
+import android.content.Context;
+import android.content.SharedPreferences;
+import android.graphics.Bitmap;
+import android.preference.PreferenceManager;
+import android.util.Log;
+import android.view.inputmethod.CursorAnchorInfo;
+import android.view.inputmethod.ExtractedText;
+import android.view.inputmethod.ExtractedTextRequest;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.annotation.UiThread;
+
+import org.mozilla.geckoview.AllowOrDeny;
+import org.mozilla.geckoview.ContentBlocking;
+import org.mozilla.geckoview.GeckoResult;
+import org.mozilla.geckoview.GeckoRuntime;
+import org.mozilla.geckoview.GeckoSession;
+import org.mozilla.geckoview.GeckoSessionSettings;
+import org.mozilla.geckoview.MediaElement;
+import org.mozilla.geckoview.WebRequestError;
+import org.mozilla.vrbrowser.R;
+import org.mozilla.vrbrowser.browser.Media;
+import org.mozilla.vrbrowser.browser.SessionChangeListener;
+import org.mozilla.vrbrowser.browser.SettingsStore;
+import org.mozilla.vrbrowser.browser.UserAgentOverride;
+import org.mozilla.vrbrowser.browser.VideoAvailabilityListener;
+import org.mozilla.vrbrowser.geolocation.GeolocationData;
+import org.mozilla.vrbrowser.telemetry.TelemetryWrapper;
+import org.mozilla.vrbrowser.utils.InternalPages;
+import org.mozilla.vrbrowser.utils.SystemUtils;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.LinkedList;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static org.mozilla.vrbrowser.utils.ServoUtils.createServoSession;
+import static org.mozilla.vrbrowser.utils.ServoUtils.isInstanceOfServoSession;
+import static org.mozilla.vrbrowser.utils.ServoUtils.isServoAvailable;
+
+public class Session implements ContentBlocking.Delegate, GeckoSession.NavigationDelegate,
+        GeckoSession.ProgressDelegate, GeckoSession.ContentDelegate, GeckoSession.TextInputDelegate,
+        GeckoSession.PromptDelegate, GeckoSession.MediaDelegate, GeckoSession.HistoryDelegate, GeckoSession.PermissionDelegate,
+        SharedPreferences.OnSharedPreferenceChangeListener {
+
+    private static final String LOGTAG = SystemUtils.createLogtag(Session.class);
+    // You can test a local file using: "resource://android/assets/webvr/index.html"
+
```

Comment:
I moved the comment next to the homepage definition. I find it useful to have the code because never remember the correct protocol path when I can to hardcode a default URL from assets